src/Orleans.Runtime/Silo/SiloHost.cs

```diff
@@ -29,19 +29,6 @@ public class SiloHost :
         /// </summary>
         public string ConfigFileName { get; set; }
 
-        /// <summary>
-        /// Directory to use for the trace log file written by this silo.
-        /// </summary>
-        /// <remarks>
-        /// <para>
-        /// The values of <c>null</c> or <c>"None"</c> mean no log file will be written by Orleans Logger manager.
-        /// </para>
-        /// <para>
-        /// When deciding The values of <c>null</c> or <c>"None"</c> mean no log file will be written by Orleans Logger manager.
-        /// </para>
-        /// </remarks>
-        public string TraceFilePath { get; set; }
```

Comment:
Looking at this made me realize that SiloHost cannot be configured to use a new filename every time.
I think that we don't need this to be configurable, but at least default to using a filename like $SiloHost-{siloName}_{DateTime.UtcNow.ToString(dateFormat)}.log".
Also, I would skip the ILoogerFactory entirely, and just create the FileLoggerProvider. Also remember to dispose the provider (or factory if you don't do the latter suggestion) when disposing this class

Reply:
There should be a way disabling the creation of log files by the silo. E.g. the location the silo is running from is read-only, or the adminstrator wants to control where and when are random files created.

Reply:
@enyim SiloHost is still there for legacy reasons. The recommendation is to move to SiloHostBuilder, and there you can fully configure logging using Microsoft.Extensions.Logging. Would that address your concern?